outputs/gcp.go

```diff
@@ -32,17 +32,27 @@ func NewGCPClient(config *types.Configuration, stats *types.Statistics, promStat
 	var storageClient *storage.Client
 
 	if config.GCP.PubSub.ProjectID != "" && config.GCP.PubSub.Topic != "" {
-		credentials, err := google.CredentialsFromJSON(context.Background(), []byte(googleCredentialsData), pubsub.ScopePubSub)
-		if err != nil {
-			log.Printf("[ERROR] : GCP PubSub - %v\n", "Error while loading GCP Credentials")
-			return nil, errors.New("Error while loading GCP Credentials")
-		}
-		pubSubClient, err := pubsub.NewClient(context.Background(), config.GCP.PubSub.ProjectID, option.WithCredentials(credentials))
-		if err != nil {
-			log.Printf("[ERROR] : GCP PubSub - %v\n", "Error while creating GCP PubSub Client")
-			return nil, errors.New("Error while creating GCP PubSub Client")
+		if googleCredentialsData != "" {
+			credentials, err := google.CredentialsFromJSON(context.Background(), []byte(googleCredentialsData), pubsub.ScopePubSub)
+			if err != nil {
+				log.Printf("[ERROR] : GCP PubSub - %v\n", "Error while loading GCP Credentials")
+				return nil, errors.New("Error while loading GCP Credentials")
+			}
+			pubSubClient, err := pubsub.NewClient(context.Background(), config.GCP.PubSub.ProjectID, option.WithCredentials(credentials))
+			if err != nil {
+				log.Printf("[ERROR] : GCP PubSub - %v\n", "Error while creating GCP PubSub Client")
+				return nil, errors.New("Error while creating GCP PubSub Client")
+			}
+			topicClient = pubSubClient.Topic(config.GCP.PubSub.Topic)
+		} else {
+			log.Printf("Workload Identity!")
```

Comment:
this is needed?

Reply:
haha no that is not. Forgot to remove that fancy debug move :) I'll clean that up today.

Reply:
Fixed!